### PR TITLE
Propose a `pdk new transport` command

### PR DESCRIPTION
**Pull request by DavidS**

### CODE REVIEW

File: RFCs/0004-add-pdk-new-transport.md [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+
+## Motivation & Goals
+
+- PDK should provide an easy way for module authors to build advanced content.
+
+- PDK should encourage module authors to adopt good development practices, such as maintaining unit test coverage of their code.
+
+## Proposed Implementation Design
+
+Implement a new pdk subcommand, `pdk new transport <transport_name>` which will render a set of templates from `object_templates/transport_*` using the provided transport name.
```
Comment (scotje) [inline RFCs/0004-add-pdk-new-transport.md]: Are there any other relevant arguments or is it just `<transport_name>`?

File: RFCs/0004-add-pdk-new-transport.md [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+
+This will build on the `pdk new provider` work for multi-file templates, and will contain the following templates:
+
+- `transport.erb`: the main file containing the implementation
+- `transport_spec.erb`: an example unit test covering the implementation skeleton
+- `transport_schema.erb`: an example schema containing default attributes recommended by us: `host`, `port`, `user`, `password`.
+
+## Unresolved Questions
+
+- like `pdk new provider` the transport templates will require the `puppet-resource_api` gem available and `mocks_with: rspec` to be set. Are we far along the route to make those two settings the default? At least for new modules?
```
Comment (scotje) [inline RFCs/0004-add-pdk-new-transport.md]: Adding the resource_api gem is probably pretty close, the `mocks_with: rspec` change requires some modifications to the template renderer...

File: RFCs/0004-add-pdk-new-transport.md (hunk 1 above)
Comment (DavidS) [reply RFCs/0004-add-pdk-new-transport.md]: I don't think so. `pdk new provider` didn't need any either.

File: RFCs/0004-add-pdk-new-transport.md (hunk 2 above)
Comment (DavidS) [reply RFCs/0004-add-pdk-new-transport.md]: In my initial implementation I've just copied the checking and messages from `pdk new provider`. I'm fine with keeping this experimental for now until the other changes can go in.

I'm also fine - if you agree - to remove the experimental label without fixing the gem and mock_with issue first.

**Comment by scotje** [comment]
Tagged as `final comment period` with intent to accept if there are no objections in the next 7 days.

**Pull request merged**